api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -144,6 +149,27 @@ public DbUser updateUserWithRetries(
     while (true) {
       dbUser = userModifier.apply(dbUser);
       dbUser = accessSyncService.updateUserAccessTiers(dbUser, agent);
+
+      boolean enableInitialCreditsExpiration =
+          configProvider.get().featureFlags.enableInitialCreditsExpiration;
+
+      if (enableInitialCreditsExpiration) {
+        List<DbAccessTier> userTiers = accessTierService.getAccessTiersForUser(dbUser);
+        Optional<DbUserInitialCreditsExpiration> maybeCreditsExpiration =
+            userInitialCreditsExpirationDao.findByUser(dbUser);
+
+        if (!userTiers.isEmpty() && maybeCreditsExpiration.isEmpty()) {
+
+          Timestamp now = clockNow();
+          Timestamp expirationTime = new Timestamp(now.getTime() + TimeUnit.DAYS.toMillis(90));
```

Comment:
Can we make 90 days as constant or probably a config?

Reply:
Good call. I added a config value.